// vue-element-admin/src/router/modules/old-menu/continous-testing.js
/** When your routing table is too long, you can split it into small modules**/

import Layout from '@/views/layout/Layout'

const continousTesting = {
  path: '/continous-testing',
  component: Layout,
  redirect: '/continous-testing/automation-usage',
  name: 'Automation',
  meta: {
    title: 'Continous Testing',
    icon: 'edit'
  },
  children: [
    {
      path: 'https://badak.gdn-app.com/index.html?page=automationCoverage',
      meta: { title: 'Automation Coverage', icon: 'link' }
    },
    /*    {
      path: 'automation-coverage',
      component: () => import('@/views/automation-coverage/index'),
      name: 'Automation Coverage',
      meta: { title: 'Automation Coverage', icon: 'link' }
    },*/
    {
      path: 'automation-usage',
      component: () => import('@/views/automation-usage/index'),
      name: 'Automation Usage',
      meta: { title: 'Automation Usage', icon: 'link' }
    },
    {
      path: 'https://badak.gdn-app.com/index.html?page=automationJobManagement',
      meta: { title: 'Automation Job Management', icon: 'link' }
    },
    {
      path: 'https://testlink.gdn-app.com/login.php',
      meta: { title: 'Test Case System', icon: 'link' }
    },
    {
      path: 'https://badak.gdn-app.com/index.html?page=crfapproval',
      meta: { title: 'CRF Approval', icon: 'link' }
    },
    {
      path: 'https://badak.gdn-app.com/index.html?page=projectgates',
      meta: { title: 'Project Gates', icon: 'link' }
    }
  ]
}

export default continousTesting
