// vue-element-admin/src/router/modules/old-menu/reporting.js
/** When your routing table is too long, you can split it into small modules**/

import Layout from '@/views/layout/Layout'

const reporting = {
  path: 'reporting',
  component: Layout,
  alwaysShow: true,
  name: 'Reporting',
  meta: {
    title: 'Reporting',
    icon: 'chart'
  },
  children: [
    {
      path: 'https://badak.gdn-app.com/index.html?page=report&type=ENG_REPORT',
      meta: { title: 'Engineering Report', icon: 'link' }
    },
    {
      path: 'https://badak.gdn-app.com/index.html?page=report&type=QIP',
      meta: { title: 'QIP', icon: 'link' }
    },
    {
      path: 'https://badak.gdn-app.com/index.html?page=cms&type=DEPT_KPI&startDate=2017-01-01&endDate=2017-12-31',
      meta: { title: 'Departement KPI', icon: 'link' }
    },
    {
      path: 'https://badak.gdn-app.com/index.html?page=cms&type=INT_QA_KPI&startDate=2017-01-01&endDate=2017-12-31',
      meta: { title: 'Internal QA KPI', icon: 'link' }
    },
    {
      path: 'https://performancekpi.gdn-app.com/job/KPI_PERFORMANCE/job/Report/HTML_Report/',
      meta: { title: 'Performance KPI Report', icon: 'link' }
    }
  ]
}

export default reporting
