// vue-element-admin/src/store/modules/automation-usage.js
import automationUsageApi from '../../api/controller/automation-usage'

const state = {
  lineChartData: {},
  collabInfo: {},
  automationUsageSummarys: {},
  projectInfo: {}
}

const mutations = {
  'SET_LINECHART_DATA': (state, payload) => {
    state.lineChartData = payload
  },
  'SET_COLLAB_INFO': (state, payload) => {
    state.collabInfo = payload
  },
  'SET_AUTOMATION_USAGE_SUMMARYS': (state, payload) => {
    state.automationUsageSummarys = payload
  },
  'SET_PROJECT_INFO': (state, payload) => {
    state.projectInfo = payload
  }
}

const actions = {
  getLineChartData: ({ commit }) => {
    return new Promise(function(resolve, reject) {
      automationUsageApi.getAutamationChartData((response) => {
        commit('SET_LINECHART_DATA', response.data.content)
        resolve(true)
      }, (error) => {
        console.log(error)
        resolve(false)
      })
    })
  },
  getCollabInfo: ({ commit }) => {
    return new Promise(function(resolve, reject) {
      automationUsageApi.getCollabInfo((response) => {
        commit('SET_COLLAB_INFO', response.data.content)
        resolve(true)
      }, (error) => {
        console.log(error)
        resolve(false)
      })
    })
  },
  getAutomationUsageSummarys: ({ commit }) => {
    return new Promise(function(resolve, reject) {
      automationUsageApi.getAutomationUsageSummary((response) => {
        commit('SET_AUTOMATION_USAGE_SUMMARYS', response.data.content)
        resolve(true)
      }, (error) => {
        console.log(error)
        resolve(false)
      })
    })
  },
  getProjectInfo: ({ commit }) => {
    return new Promise(function(resolve, reject) {
      automationUsageApi.getProjectInfo((response) => {
        commit('SET_PROJECT_INFO', response.data.content)
        resolve(true)
      }, (error) => {
        console.log(error)
        resolve(false)
      })
    })
  }
}

const getters = {
  lineChartData: (state) => {
    return state.lineChartData
  },
  collabInfo: (state) => {
    return state.collabInfo
  },
  automationUsageSummarys: (state) => {
    return state.automationUsageSummarys
  },
  projectInfo: (state) => {
    return state.projectInfo
  }
}

const namespaced = true

export default {
  state,
  mutations,
  actions,
  getters,
  namespaced
}
