// vue-element-admin/src/router/index.js
import Vue from 'vue'
import Router from 'vue-router'

Vue.use(Router)

/* Layout */
import Layout from '@/views/layout/Layout'
import continousTesting from './modules/old-menu/continous-testing'
import continousDelivery from './modules/old-menu/continous-delivery'
import smartTools from './modules/old-menu/smart-tools'
import smartResources from './modules/old-menu/smart-resources'
import reporting from './modules/old-menu/reporting'
import userManagement from './modules/old-menu/user-management'
import projectManagement from './modules/old-menu/project-management'
import accountSetting from './modules/old-menu/account-setting'
import systemSetup from './modules/old-menu/system-setup'

/** note: Submenu only appear when children.length>=1
 *  detail see  https://panjiachen.github.io/vue-element-admin-site/guide/essentials/router-and-nav.html
 **/

export const constantRouterMap = [
  {
    path: '/',
    hidden: false,
    redirect: '/continous-testing/automation-usage',
    component: Layout,
    children: [
      {
        name: 'dashboard',
        path: 'https://badak.gdn-app.com/index.html?page=dashboard',
        meta: { title: 'Home', icon: 'dashboard' }
      }
    ]
  },
  {
    path: '/redirect',
    component: Layout,
    hidden: true,
    children: [
      {
        path: '/redirect/:path*',
        component: () => import('@/views/redirect/index')
      }
    ]
  },
  continousDelivery,
  continousTesting,
  smartTools,
  smartResources,
  reporting,
  userManagement,
  projectManagement,
  systemSetup,
  accountSetting,
  {
    path: '/404',
    component: () => import('@/views/errorPage/404'),
    hidden: true
  },
  {
    path: '/401',
    component: () => import('@/views/errorPage/401'),
    hidden: true
  }
/*  {
    path: '*',
    component: Layout,
    redirect: '/404',
    hidden: true
  }*/
]

export default new Router({
  // mode: 'history', // require service support
  scrollBehavior: () => ({ y: 0 }),
  routes: constantRouterMap
})
